File: PasswordValidation/PasswordValidation/PasswordValidatorBuilder.cs
namespace PasswordValidation;

public class PasswordValidatorBuilder : IPasswordValidatorBuilder
{
    private readonly List<IPasswordValidator> _validators = new List<IPasswordValidator>();
    
    public IPasswordValidatorBuilder Add(IPasswordValidator validator)
    {
        _validators.Add(validator);
        return this;
    }

    public CompositePasswordValidator Build()
    {
        return new CompositePasswordValidator(_validators);
    }
}

File: PasswordValidation/py/password_validation/password_validator_builder.py
from password_validation.composite_password_validator import CompositePasswordValidator
from password_validation.flexible_password_validator import PasswordValidator


class PasswordValidatorBuilder:
    def __init__(self):
        self.__validators: list = []

    def add(self, validator: PasswordValidator):
        self.__validators.append(validator)
        return self

    def build(self):
        return CompositePasswordValidator(self.__validators)
